File: problem1/problem1.clj
(defn two-sum
  "Loop through xs. Insert the complement for each x into map as {x index}.
   If the complement for a given x is found, return the index of the x and the complement."
  [xs target]
  (loop [comps {} ; map of complements and indexes
         idx 0 ; current index
         comp (- target (nth xs idx))] ; current complement
    (if (contains? comps comp)
      [(get comps comp) idx]
      (recur
       (assoc comps comp idx)
       (inc idx)
       (- target (nth xs idx))))))

(print (two-sum [2 7 11 15] 9))

File: problem1/problem1.py
from typing import List

def twoSum(nums: List[int], target: int) -> List[int]:
    # Loop through each number in list
    for idx, num in enumerate(nums):
        # Test adding rest of numbers in list to the given number
        # Start the second loop after the index of the given number
        # to avoid duplicate work
        for jdx, comp in enumerate(nums[idx+1:]):
            if target == comp + num:
                return [idx, jdx + idx + 1]

print(twoSum([2, 7, 11, 15], 9) == [0, 1])